pkg/kubernetes/service/v1alpha1/build.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Builder is the builder object for Service
+type Builder struct {
+	service *Service
+	errs    []error
+}
+
+// NewBuilder returns new instance of Builder
+func NewBuilder() *Builder {
+	return &Builder{service: &Service{object: &corev1.Service{}}}
+}
+
+// WithName sets the Name field of Service with provided value.
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build service object: missing service name"))
+		return b
+	}
+	b.service.object.Name = name
+	return b
+}
+
+// WithGenerateName sets the GenerateName field of
+// Service with provided value
+func (b *Builder) WithGenerateName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build service object: missing service generateName"),
+		)
+		return b
+	}
+
+	b.service.object.GenerateName = name
+	return b
+}
+
+// WithNamespace sets the Namespace field of Service provided arguments
+func (b *Builder) WithNamespace(namespace string) *Builder {
+	if len(namespace) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build service object: missing service namespace"),
+		)
+		return b
+	}
+	b.service.object.Namespace = namespace
+	return b
+}
+
+// WithAnnotations sets the Annotations field of Service with provided arguments
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build service object: missing annotations"))
+		return b
+	}
+
+	// copy of original map
+	newannotations := map[string]string{}
+	for key, value := range annotations {
+		newannotations[key] = value
+	}
+
+	// override
+	b.service.object.Annotations = newannotations
```

Comment:
This should not be override. It should be merge.
WithAnnotationsNew can be another build method that overrides

Reply:
done